File: tests/fixtures/go-example/strings/strings.go
// Package strings contains pure string-manipulation functions used as
// fixtures for the Regrets Go capture/validate flow.
package strings

// Reverse returns the input string with its runes reversed.
func Reverse(s string) string {
	r := []rune(s)
	for i, j := 0, len(r)-1; i < j; i, j = i+1, j-1 {
		r[i], r[j] = r[j], r[i]
	}
	return string(r)
}

// CountVowels returns the number of vowels (a, e, i, o, u) in the input,
// case-insensitively.
func CountVowels(s string) int {
	count := 0
	for _, c := range s {
		switch c {
		case 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U':
			count++
		}
	}
	return count
}

// IsPalindrome returns true if the input reads the same forwards and backwards.
func IsPalindrome(s string) bool {
	return s == Reverse(s)
}

File: tests/fixtures/lua-example/strings.lua
-- strings.lua — example Lua module for the Regrets Lua stack
--
-- Three pure functions used by tests/fixtures/lua-example/regrets/manifest.json
-- clusters. Pure: no I/O, no globals, no side effects — deterministic output
-- for a given input. Ideal fingerprint targets.

local M = {}

-- Reverse a string.
-- e.g. M.reverse("hello") == "olleh"
function M.reverse(s)
    return string.reverse(s)
end

-- Count vowels (a, e, i, o, u — case-insensitive) in a string.
-- e.g. M.count_vowels("hello") == 2
function M.count_vowels(s)
    local _, n = string.gsub(s, "[aeiouAEIOU]", "")
    return n
end

-- Is the string a palindrome? (reads the same forwards and backwards)
-- e.g. M.is_palindrome("level") == true
function M.is_palindrome(s)
    return s == M.reverse(s)
end

return M
